src/main/java/io/horrorshow/codey/parser/TimeParser.java

```diff
@@ -0,0 +1,89 @@
+package io.horrorshow.codey.parser;
+
+import java.text.DateFormatSymbols;
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoField;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+
+public class TimeParser {
+    public static OffsetDateTime toOffsetDateTime(String group) {
```

Comment:
lots of the content in this class can be simplified, since there are many unnecessary steps like the splitting of group again and again instead of splitting it once and putting the parts in a variable. Since the regex matched by the input before it comes here guarantees that, if there is a weekday it's a valid one, so we don't need the `weekDays`-List anymore to check if it's an actual properly spelled day.

After removing the unnecessary steps the remaining steps of this class fit ok'ish in this one method:
```java
    public static OffsetDateTime toOffsetDateTime(String group) {
        var parts = group.split("\\s+");
        var hasDay = parts.length == 3;
        var HHmm = parts[hasDay ? 1 : 0].split(":");
        var region = parts[hasDay ? 2 : 1];

        var offsetDateTime = OffsetDateTime.now(ZoneId.of(region))
                .withHour(Integer.parseInt(HHmm[0]))
                .withMinute(Integer.parseInt(HHmm[1]));
        if (hasDay) {
            var dayOfTheWeek = DayOfWeek.valueOf(parts[0].toUpperCase(Locale.ROOT));
            offsetDateTime = offsetDateTime.with(ChronoField.DAY_OF_WEEK, dayOfTheWeek.getValue());
        }
        return offsetDateTime;
    }
```